api/src/main/java/org/openmrs/module/attachments/obs/ImageAttachmentHandler.java

```diff
@@ -65,21 +71,30 @@ protected boolean deleteComplexData(Obs obs, AttachmentComplexData complexData)
 	
 	@Override
 	protected ValueComplex saveComplexData(Obs obs, AttachmentComplexData complexData) {
+		int imageHeight = Integer.MAX_VALUE;
+		int imageWidth = Integer.MAX_VALUE;
+		String savedFileName = null;
 		
 		// We invoke the parent to inherit from the file saving routines.
 		obs = getParent().saveObs(obs);
 		
 		File savedFile = AbstractHandler.getComplexDataFile(obs);
-		String savedFileName = savedFile.getName();
 		
-		// Saving the thumbnail
-		File dir = savedFile.getParentFile();
-		String thumbnailFileName = buildThumbnailFileName(savedFileName);
+		// Get image dimensions
 		try {
-			Thumbnails.of(savedFile.getAbsolutePath()).size(THUMBNAIL_MAX_HEIGHT, THUMBNAIL_MAX_WIDTH)
-			        .toFile(new File(dir, thumbnailFileName));
+			BufferedImage image = ImageIO.read(savedFile);
+			imageHeight = image.getHeight();
+			imageWidth = image.getWidth();
 		}
 		catch (IOException e) {
+			log.warn(
+			    "The dimensions of image 'savedFileName' could not be determined, continuing with generating thumbnail.");
```

Comment:
```java
"The dimensions of image file '" + savedFileName + "' could not be determined, continuing with generating its thumbnail anyway.");
```
I rephrased a little but mostly I made sure that the actual file name becomes part of the logged message by concatenating the String variable `savedFileName`.